models/StallPayment.js

```diff
@@ -1,10 +1,13 @@
-import * as mongoose from "mongoose";
-import { Schema } from "mongoose";
+import * as mongoose from 'mongoose';
+import { Schema } from 'mongoose';
 
 const StallPaymentSchema = new Schema(
   {
-    vendorId: String,
-    stallId: String,
+    vendorId: {
+      type: mongoose.Types.ObjectId,
+      ref: 'User',
+    },
+    stallId: String, // not sure which this should be
```

Comment:
EventStall

Reply:
Wait, no... I don't see EventStall in the models

Reply:
Its inside the Event file. I wanted to make an array of EventStalls for the stall fileld. I saw something similar in the mongoose docs

Reply:
Like how they did it here

https://mongoosejs.com/docs/schematypes.html#arrays

Reply:
Slightly lost... If it isn't a mongoose object with a schema, i think we can merge as is.
But make the change if you know exactly what's what.

Reply:
Oh yea. I was saying that the EventStall schema is inside the event file. I wanted to do what the link shows. 

As for the Id it just becomes {
      type: mongoose.Types.ObjectId,
      ref: 'EventStall',
    },
    
   Unless the ref is for the file?

Reply:
May i have the file path to the events file.

I only see events.js

Reply:
Mara it doesnt look like it. Can i please get a link for more info on refs?

Reply:
But your logic would be sound if it is a proper schema

Reply:
This one? http://localhost:3000/vendor/events/1

Reply:
vendor/ 

displays the index which shows all events